src/Input/Input.js

```diff
@@ -130,7 +130,7 @@ export default class Input extends Component {
     /**
      * The input value, required for a controlled component.
      */
-    value: PropTypes.string,
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```

Comment:
Oh, I didn't know we had `stringOrNumber`. I would rather remove that helper.
I don't think that this abstraction add much value.